src/backend/distributed/executor/multi_utility.c

```diff
@@ -850,6 +863,30 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 		{
 			case AT_AddColumn:
 			{
+				if (IsA(command->def, ColumnDef))
+				{
+					ColumnDef *column = (ColumnDef *) command->def;
+
+					/* Check for SERIAL pseudo-types */
+					if (column->typeName && list_length(column->typeName->names) == 1 &&
```

Comment:
Is that list_length() check correct? pg_catalog.serial is sometimes auto-generated.